go/oasis-node/cmd/ias/auth_registry.go

```diff
@@ -53,15 +53,40 @@ func (auth *registryAuthenticator) VerifyEvidence(ctx context.Context, evidence
 	return nil
 }
 
-func (auth *registryAuthenticator) worker(ctx context.Context) {
-	defer auth.conn.Close()
+func (auth *registryAuthenticator) dialRegistry(ctx context.Context) (*grpc.ClientConn, error) {
+	conn, err := cmdGrpc.NewClient(auth.cmd)
+	if err != nil {
+		return nil, errors.Wrap(err, "ias: failed to create gRPC client")
+	}
+	return conn, nil
+}
 
+func (auth *registryAuthenticator) worker(ctx context.Context) {
 	waitRuntimes := viper.GetInt(cfgWaitRuntimes)
 	if waitRuntimes <= 0 {
 		close(auth.initCh)
 	}
 
-	ch, sub, err := auth.client.WatchRuntimes(ctx)
+	var redialAttempts uint
+
+Redial:
```

Comment:
Could use the [cenkalti/backoff](https://github.com/cenkalti/backoff) package we use in other similar cases?

Reply:
I think that would be an overkill for this simple use case -- the only time this connection is dropped is if the upstream node rotates its TLS certificates, a constant 2s retry is good enough.